### PR TITLE
[Improvement] Correct tests handling for EagerService

### PR DESCRIPTION
**Pull request by mproch**
The main change is passing TestRunId to compiler explicitly, without hacks in e.g. TestFlinkProcessCompiler. 
This is first PR with test handling refactor. Some more will follow :)

### CODE REVIEW

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/test/InvocationCollectors.scala [hunk 1]
```diff
@@ -94,10 +93,10 @@ object InvocationCollectors {
 
   }
 
-  case class TestServiceInvocationCollector private(runIdOpt: Option[TestRunId],
+  case class TestServiceInvocationCollector(runIdOpt: Option[TestRunId],
                                                     contextId: ContextId,
```
Comment (lciolecki) [inline engine/api/src/main/scala/pl/touk/nussknacker/engine/api/test/InvocationCollectors.scala:97]: Ugly formatting

File: engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/helpers/SampleNodes.scala [hunk 2]
```diff
@@ -191,6 +191,23 @@ object SampleNodes {
 
   }
 
+  object CollectingEagerService extends EagerService {
+
+    @MethodToInvoke
+    def invoke(@ParamName("static") static: String,
```
Comment (lciolecki) [inline engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/helpers/SampleNodes.scala:197]: Can you do declaring invoke in one line? :)

**Review by lciolecki — APPROVED** [review]
LGTM

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/test/InvocationCollectors.scala (hunk 1 above)
Comment (mproch) [reply engine/api/src/main/scala/pl/touk/nussknacker/engine/api/test/InvocationCollectors.scala]: fixed...

File: engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/helpers/SampleNodes.scala (hunk 2 above)
Comment (mproch) [reply engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/helpers/SampleNodes.scala]: ok...

**Pull request merged**